core/src/main/java/io/questdb/griffin/engine/functions/date/TimestampAddFunctionFactory.java

```diff
@@ -51,125 +51,127 @@ public String getSignature() {
     }
 
     @Override
-    public Function newInstance(int position, ObjList<Function> args, IntList argPositions, CairoConfiguration configuration, SqlExecutionContext sqlExecutionContext) {
-
-        Function period = args.getQuick(0);
-        Function interval = args.getQuick(1);
-        if (period.isConstant()) {
-            char periodValue = period.getChar(null);
-            if (periodValue < addFunctionsMax) {
-                LongAddIntFunction func = addFunctions.getQuick(periodValue);
-                if (func != null) {
-                    if (interval.isConstant()) {
-                        if (interval.getInt(null) != Numbers.INT_NULL) {
-                            return new AddLongIntVarConstFunction(args.getQuick(2), interval.getInt(null), func, periodValue);
-                        }
-                        return TimestampConstant.NULL;
+    public Function newInstance(int position, ObjList<Function> args, IntList argPositions, CairoConfiguration configuration, SqlExecutionContext sqlExecutionContext) throws SqlException {
+        Function periodFunc = args.getQuick(0);
+        Function strideFunc = args.getQuick(1);
+        Function timestampFunc = args.getQuick(2);
+        LongAddIntFunction periodAddFunc;
+        char period;
+        int stride;
+
+        if (periodFunc.isConstant()) {
+            if ((period = periodFunc.getChar(null)) < addFunctionsMax && (periodAddFunc = addFunctions.getQuick(period)) != null) {
```

Comment:
the code was good, i was asking for a comment to explain that "addFunction" is sparse data structure, which is why we're comparing "character" to the data structure size etc.

Reply:
Currently we have a LUT (lookup table). It may be faster than a switch especially JVM's `tableswitch`, not `lookupswitch`, but on the other hand it makes the code less readable, so to me we need a good reason to keep it.